server/service/queries.go

```diff
@@ -86,30 +113,47 @@ func listQueriesEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 	for _, query := range queries {
 		respQueries = append(respQueries, *query)
 	}
+
 	return listQueriesResponse{
 		Queries: respQueries,
+		Count:   count,
+		Meta:    meta,
 	}, nil
 }
 
-func (svc *Service) ListQueries(ctx context.Context, opt fleet.ListOptions, teamID *uint, scheduled *bool, mergeInherited bool) ([]*fleet.Query, error) {
+func (svc *Service) ListQueries(ctx context.Context, opt fleet.ListOptions, teamID *uint, scheduled *bool, mergeInherited bool, compatiblePlatform *string) ([]*fleet.Query, int, *fleet.PaginationMetadata, error) {
 	// Check the user is allowed to list queries on the given team.
 	if err := svc.authz.Authorize(ctx, &fleet.Query{
 		TeamID: teamID,
 	}, fleet.ActionRead); err != nil {
-		return nil, err
+		return nil, 0, nil, err
 	}
 
-	queries, err := svc.ds.ListQueries(ctx, fleet.ListQueryOptions{
+	// always include metadata for queries
+	opt.IncludeMetadata = true
+
+	queries, count, meta, err := svc.ds.ListQueries(ctx, fleet.ListQueryOptions{
 		ListOptions:    opt,
 		TeamID:         teamID,
 		IsScheduled:    scheduled,
 		MergeInherited: mergeInherited,
 	})
 	if err != nil {
-		return nil, err
+		return nil, 0, nil, err
+	}
+
+	if compatiblePlatform != nil {
+		// TODO on future iteration - instead of filtering per call like this, calculate each query's compatible platform
+		// and store it in the db on save/edit (query.compatible_platform, different from
+		// query.platform), then update svc.ds.ListQueries to filter by compatible platform.
+		// This avoids overhead of a migration for now
```

Comment:
What is this `compatible_platform`? Isn't it a UI thing only when editing / creating a SQL?

WE should always use `query.platform` to know exactly where the query will run.

Reply:
Why do we want to filter by "Compatible platform" and not "platform"?

As a user, why do I want to know if a query that I wrote for Linux devices is compatible with macOS?

Reply:
@lucasmrod since we discussed all of this in our discussion earlier today I'm going to mark this as resolved. Please re-open if I missed something!